src/material/datepicker/calendar.ts

```diff
@@ -472,3 +480,18 @@ export class MatCalendar<D> implements AfterContentInit, AfterViewChecked, OnDes
     return this.monthView || this.yearView || this.multiYearView;
   }
 }
+
+/**
+ * Function that provides the next calendar view according to the current view.
+ * @docs-private
+ */
+function _getNextCalendarView(
+  currentView: MatCalendarView,
+  isStartViewMultiYear: boolean,
+): MatCalendarView {
+  return {
```

Comment:
This is super inefficient because every invocation is going to create an object and a bunch of function just so it can map to the correct value. If we end up doing it, we should just change it to a `switch` or some `else if` statements.

Reply:
These changes are no longer needed (. But I agree about switch\if statement.